rest/server_context_test.go

```diff
@@ -163,3 +165,187 @@ func (client *MockClient) RespondToGET(url string, response *http.Response) {
 	tripper := client.Transport.(*mockTripper)
 	tripper.getURLs[url] = response
 }
+
+func TestAllDatabaseNames(t *testing.T) {
+	server := "walrus:"
+	bucketName := "imbucket"
+
+	serverConfig := &ServerConfig{CORS: &CORSConfig{}, AdminInterface: &DefaultAdminInterface}
+	serverContext := NewServerContext(serverConfig)
+	bucketConfig := BucketConfig{Server: &server, Bucket: &bucketName}
+
+	dbConfig := &DbConfig{BucketConfig: bucketConfig, Name: "imdb1", AllowEmptyPassword: true}
+	dbContext, err := serverContext.AddDatabaseFromConfig(dbConfig)
+	assert.NoError(t, err, "No error while adding database to server context")
+	assert.Equal(t, server, dbContext.BucketSpec.Server)
+	assert.Equal(t, bucketName, dbContext.BucketSpec.BucketName)
+	assert.Len(t, serverContext.AllDatabaseNames(), 1)
+	assert.Contains(t, serverContext.AllDatabaseNames(), "imdb1")
+
+	dbConfig = &DbConfig{BucketConfig: bucketConfig, Name: "imdb2", AllowEmptyPassword: true}
+	dbContext, err = serverContext.AddDatabaseFromConfig(dbConfig)
+	assert.NoError(t, err, "No error while adding database to server context")
+	assert.Equal(t, server, dbContext.BucketSpec.Server)
+	assert.Equal(t, bucketName, dbContext.BucketSpec.BucketName)
+	assert.Len(t, serverContext.AllDatabaseNames(), 2)
+	assert.Contains(t, serverContext.AllDatabaseNames(), "imdb1")
+	assert.Contains(t, serverContext.AllDatabaseNames(), "imdb2")
+
+	status := serverContext.RemoveDatabase("imdb2")
+	assert.True(t, status, "Database should be removed from server context")
+	assert.Len(t, serverContext.AllDatabaseNames(), 1)
+	assert.Contains(t, serverContext.AllDatabaseNames(), "imdb1")
+	assert.NotContains(t, serverContext.AllDatabaseNames(), "imdb2")
+	serverContext.Close()
+}
+
+func TestStartReplicators(t *testing.T) {
```

Comment:
I think we should remove this test until we have a way to run an actual sg-replicate replication from the unit test framework.  Until then, the first three cases are already handled by validateReplicationParameters coverage elsewhere, and the last one is going to be race-y.

Reply:
Removed the test in latest revision.